discovery_engine/lib/src/domain/event_handler.dart

```diff
@@ -286,7 +292,12 @@ class EventHandler {
       activeDataRepository,
       engineStateRepository,
     );
-    _systemManager = SystemManager(engine, eventConfig, documentRepository);
+    _systemManager = SystemManager(
+      engine,
+      eventConfig,
+      documentRepository,
+      aiStateHolders,
```

Comment:
:grey_question: why passing this list instead of a function to call when we want to delete the state?
Here we know about hive so we can create it directly on the box that are involved and we don't expose `clearAIState` on each repository. We will also avoid that on `clearAIState` is called on some repository and not others when it only make sense when is called on all of them.

Reply:
callbacks don't scale well, but I guess in this case we could do it

Reply:
In this case the implementation of each `clearAIState` is one line so is not much of a problem. Also we are trying to move away from dart so I would like to minimize changes here when possibile.

Reply:
yes, exactly I just wanted to explain how I ended up where I ended up :wink: